Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.Events.cs

```diff
@@ -21,13 +22,61 @@ namespace Microsoft.Toolkit.Uwp.UI.Controls
     public partial class InAppNotification
     {
         /// <summary>
+        /// Event raised when the notification is opening
+        /// </summary>
+        public event InAppNotificationOpeningEventHandler Opening;
+
+        /// <summary>
+        /// Event raised when the notification is opened
+        /// </summary>
+        public event EventHandler Opened;
+
+        /// <summary>
+        /// Event raised when the notification is dismissing
+        /// </summary>
+        [Obsolete("Use Closing event instead.")]
+        public event InAppNotificationDismissingEventHandler Dismissing;
+
+        /// <summary>
         /// Event raised when the notification is dismissed
         /// </summary>
+        [Obsolete("Use Closed event instead.")]
         public event EventHandler Dismissed;
 
+        /// <summary>
+        /// Event raised when the notification is closing
+        /// </summary>
+        public event InAppNotificationClosingEventHandler Closing;
+
+        /// <summary>
+        /// Event raised when the notification is closed
+        /// </summary>
+        public event EventHandler Closed;
```

Comment:
Should the Closed EventHandler also provide the details on how it was closed?

Reply:
@nmetulev Well, it can a little bit harder for this part. I used a Timer for both Opened and Closed events, started after the Opening and Closing events respectively. I am not sure if it's possible but I can eventually store the `InAppNotificationDismissKind` before Tick event. What do you think?

Reply:
Yeah, that makes sense to me, you can store the last dismisskind in a variable